source/Cosmos.Core/GCImplementation.cs

```diff
@@ -1,66 +1,126 @@
+    {
+        private unsafe static byte* memPtr = null;
+        private static uint memLength = 0;
+        /// <summary>
+        /// Acquire lock. Not implemented.
+        /// </summary>
+        /// <exception cref="NotImplementedException">Thrown always.</exception>
+        private static void AcquireLock()
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Release lock. Not implemented.
+        /// </summary>
+        /// <exception cref="NotImplementedException">Thrown always.</exception>
+        private static void ReleaseLock()
+        {
+            throw new NotImplementedException();
+        }
+        /// <summary>
+        /// Alloc new object.
+        /// </summary>
+        public unsafe static uint AllocNewObject(uint aSize)
+        {
+              return (uint)Memory.Heap.Alloc(aSize);
+            
+        }
+        /// <summary>
+        /// Free Object from Memory
+        /// </summary>
+        /// <param name="obj">Takes a memory allocated object</param>
+        public unsafe static void Free(object aObj)
+        {
+            Memory.Heap.Free(GetPointer(aObj));
+        }
+
+        /// <summary>
+        /// Increase reference count of an given object. Plugged.
+        /// </summary>
+        /// <param name="aObject">An object to increase to reference count of.</param>
+        /// <exception cref="NotImplementedException">Thrown on fatal error, contact support.</exception>
+        public static unsafe void IncRefCount(uint aObject)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Decrease reference count of an given object. Plugged.
+        /// </summary>
+        /// <param name="aObject">An object to decrease to reference count of.</param>
+        /// <exception cref="NotImplementedException">Thrown on fatal error, contact support.</exception>
+        public static unsafe void DecRefCount(uint aObject)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Get amount of available Ram
+        /// </summary>
+        /// <returns>Returns amount of available memory to the System in MB</returns>
+        public static uint GetAvailableRAM()
+        {
+            return memLength / 1024 / 1024;
+        }
+        /// <summary>
+        /// Get a rough estimate of used Memory by the System
+        /// </summary>
+        /// <returns>Returns the used PageSize by the MemoryManager in Bytes.</returns>
+        public static uint GetUsedRAM()
+        {
+            return (Memory.RAT.TotalPageCount - Memory.RAT.GetPageCount(Memory.RAT.PageType.Empty)) * Memory.RAT.PageSize;
+        }
+        /// <summary>
+        /// Initialise the Memory Manager, this should not be called anymore since it is done very early during the boot process.
+        /// </summary>
+        public static unsafe void Init()

```

Comment:
may want to make this internal not public since it should not be called from out side of Cosmos's boot precess

Reply:
Making this private would prevent it from being called from the boot process as well considering that the bootstrap is not in this class

Reply:
if it was internal (can only be call by code in the same dll) it can still be called